### ISSUE TITLE
FI-TSNE 0.1.4 Fails on FlowJo Workflow.

### ISSUE THREAD

**Issue by martin1english** [tercen/app-library#3]
Workflow
https://stage.tercen.com/TestTeam/w/a33e2c0f05753e1b035d27049abb9871

Tested on Apple Mac - Safari

Error Message

> Unexpected error : run.operator.exit.code.1
run.operator.exit.code.1 : Failed to run operator, exit code = 1
--OUT--
=============== t-SNE v1.2.1 ===============
fast_tsne data_path: /tmp/RtmpOYRgGS/fftRtsne_data_97b48ff4f.dat
fast_tsne result_path: /tmp/RtmpOYRgGS/fftRtsne_result_9799d940f.dat
fast_tsne nthreads: 16
Read the following parameters:
n 1542846 by d 8 dataset, theta 0.500000,
perplexity 30.000000, no_dims 2, max_iter 750,
stop_lying_iter 250, mom_switch_iter 250,
momentum 0.500000, final_momentum 0.800000,
learning_rate 128570.500000, max_step_norm 5.000000,
K -1, sigma -30.000000, nbody_algo 2,
knn_algo 1, early_exag_coeff 12.000000,
no_momentum_during_exag 0, n_trees 50, search_k 4500,
start_late_exag_iter -1, late_exag_coeff 1.000000
nterms 3, interval_per_integer 1.000000, min_num_intervals 50, t-dist df 1.000000
Read the 1542846 x 8 data matrix successfully. X[0,0] = 251.000000
Read the initialization successfully.
Will use momentum during exaggeration phase
Computing input similarities...
Using perplexity, so normalizing input data (to prevent numerical problems)
Using perplexity, not the manually set kernel width. K (number of nearest neighbors) and sigma (bandwidth) parameters are going to be ignored.
Using ANNOY for knn search, with parameters: n_trees 50 and search_k 4500
Going to allocate memory. N: 1542846, K: 90, N*K = 138856140
Building Annoy tree...
Done building tree. Beginning nearest neighbor search... 
parallel (16 threads):
[> ] 0% 0.164s
[> ] 0% 1.597s
[> ] 0% 3.054s
[> ] 0% 4.583s
[> ] 0% 6.057s
[> ] 0% 7.506s
[> ] 0% 9.012s
[> ] 0% 10.426s
[> ] 0% 11.895s
[> ] 0% 13.359s
[> ] 1% 14.866s
[> ] 1% 16.364s
[> ] 1% 17.835s
[> ] 1% 19.329s
[> ] 1% 20.795s
[> ] 1% 22.2s
[> ] 1% 23.792s
[=> ] 1% 25.278s
[=> ] 1% 26.708s
[=> ] 1% 28.167s
[=> ] 2% 29.639s
[=> ] 2% 31.137s
[=> ] 2% 32.677s
[=> ] 2% 34.197s
[=> ] 2% 35.773s
[=> ] 2% 37.309s
[=> ] 2% 38.845s
[=> ] 2% 40.398s
[=> ] 2% 41.922s
[=> ] 2% 43.461s
[=> ] 3% 44.965s
[=> ] 3% 46.425s
[=> ] 3% 47.894s
[=> ] 3% 49.491s
[==> ] 3% 50.997s
[==> ] 3% 52.531s
[==>

**Comment by martin1english** [comment]
@fnaji @fpnaji also testing on PC.

**Comment by fnaji** [comment]
My run failed. It takes 30-40 mins.

Here is the log

n 1542846 by d 8 dataset, theta 0.500000,
perplexity 30.000000, no_dims 2, max_iter 750,
stop_lying_iter 250, mom_switch_iter 250,
momentum 0.500000, final_momentum 0.800000,
learning_rate 128570.500000, max_step_norm 5.000000,
K -1, sigma -30.000000, nbody_algo 2,
knn_algo 1, early_exag_coeff 12.000000,
no_momentum_during_exag 0, n_trees 50, search_k 4500,
start_late_exag_iter -1, late_exag_coeff 1.000000
nterms 3, interval_per_integer 1.000000, min_num_intervals 50, t-dist df 1.000000
Read the 1542846 x 8 data matrix successfully. X[0,0] = 251.000000
Read the initialization successfully.
Will use momentum during exaggeration phase
Computing input similarities...
Using perplexity, so normalizing input data (to prevent numerical problems)
Using perplexity, not the manually set kernel width. K (number of nearest neighbors) and sigma (bandwidth) parameters are going to be ignored.
Using ANNOY for knn search, with parameters: n_trees 50 and search_k 4500
Going to allocate memory. N: 1542846, K: 90, N*K = 138856140
Building Annoy tree...
Done building tree. Beginning nearest neighbor search...
parallel (16 threads):

**Comment by agouy** [comment]
@fnaji @martin1english Please try again with version 0.1.4 (part of the app library). I have suppressed messages so logs are less busy as we are missing the error from the logs. I have also manually increased memory resources for this operator from version 0.1.4.

If this does not solve the issue, I see that the operator tries to apply some memory settings - it might overwrite Tercen's ones, which would explain why we are having more trouble with this operator. Please let me know if you still observe any issue with 0.1.4

**Comment by fpnaji** [comment]
The logs you see are using v 0.1.4,  Martin as well.

**Comment by agouy** [comment]
> The logs you see are using v 0.1.4,  Martin as well.

Ha I see what happens. I suppressed R messages but the log comes from the standard output of the program we call. I'll fix that.

It seems to work with new mem settings on Martin's workflow
